poplar_recovery_builder.sh

```diff
@@ -100,35 +100,45 @@ function parseargs() {
 	# Make sure a single argument was supplied
 	[ $# -lt 1 ] && usage "no partition specified"
 
+	IMAGE_TYPE=Linux
 	INPUT_FILES="L_LOADER USB_LOADER"
 	INPUT_FILES="${INPUT_FILES}"
 
-	if [ "$1" = "all" ] || [ "$1" = "system" ] ; then
-		[ -z "$2" ] && usage "no arg (filesystem) supplied"
-		ROOT_FS_ARCHIVE=$2
-	fi
-
 	case $1 in
-	all|layout|loader|boot|system)
+	all|system)
+		#echo "case all|system" >&2
+		[ -z "$2" ] && usage "no arg (filesystem) supplied"
+		if [ "$2" = "android" ]; then
+			IMAGE_TYPE=Android
+			INPUT_FILES="${INPUT_FILES} ANDROID_BOOT_IMAGE"
+			INPUT_FILES="${INPUT_FILES} ANDROID_SYSTEM_IMAGE"
+			INPUT_FILES="${INPUT_FILES} ANDROID_CACHE_IMAGE"
+			INPUT_FILES="${INPUT_FILES} ANDROID_USER_DATA_IMAGE"
+		else
+			ROOT_FS_ARCHIVE=$2
+			INPUT_FILES="${INPUT_FILES} ROOT_FS_ARCHIVE"
+		fi
+		;;&
+	all|boot)
+		#echo "case all|boot" >&2
+		INPUT_FILES="${INPUT_FILES} KERNEL_IMAGE"
+		INPUT_FILES="${INPUT_FILES} DEVICE_TREE_BINARY"
+		;;&
+	boot|loader|layout)
```

Comment:
I guess providing "all loader fs_image.tgz" will produce an error.  Anyway, no big deal, I think what you're doing here to parse arguments is good.

Reply:
Right. The rootfs must follow `all` so we're looking for a file called `loader` which presumably doesn't exist. To allow random combinations of partitions it might be better to move to using switches (e.g. `-a arg`, `--set foo`, etc) but that's probably a story for another day?